NitroxClient/GameLogic/Spawning/Metadata/SubNameInputProcessor.cs

```diff
@@ -0,0 +1,33 @@
+using NitroxModel.DataStructures.GameLogic.Entities.Metadata;
+using NitroxModel_Subnautica.DataStructures;
+using UnityEngine;
+
+namespace NitroxClient.GameLogic.Spawning.Metadata;
+
+public class SubNameInputProcessor : GenericEntityMetadataProcessor<SubNameInputMetadata>
+{
+    public override void ProcessMetadata(GameObject gameObject, SubNameInputMetadata metadata)
```

Comment:
Not sure where this could come from but this error happened when I tried rejoining a session (which I left while standing doing nothing in a cyclops)
```cs
[18:19:25.706] [ERR-UNITY] NullReferenceException: Object reference not set to an instance of an object
NitroxClient.GameLogic.Spawning.Metadata.SubNameInputProcessor.ProcessMetadata (UnityEngine.GameObject gameObject, NitroxModel.DataStructures.GameLogic.Entities.Metadata.SubNameInputMetadata metadata) (at <3371c3bf515d4df883a718364c0f20e1>:0)
NitroxClient.GameLogic.Spawning.Metadata.GenericEntityMetadataProcessor`1[T].ProcessMetadata (UnityEngine.GameObject gameObject, NitroxModel.DataStructures.GameLogic.Entities.Metadata.EntityMetadata metadata) (at <3371c3bf515d4df883a718364c0f20e1>:0)
NitroxClient.GameLogic.Spawning.Metadata.EntityMetadataProcessor.ApplyMetadata (UnityEngine.GameObject gameObject, NitroxModel.DataStructures.GameLogic.Entities.Metadata.EntityMetadata metadata) (at <3371c3bf515d4df883a718364c0f20e1>:0)
NitroxClient.GameLogic.Entities+<SpawnAsync>d__13.MoveNext () (at <3371c3bf515d4df883a718364c0f20e1>:0)
UnityEngine.SetupCoroutine.InvokeMoveNext (System.Collections.IEnumerator enumerator, System.IntPtr returnValueAddress) (at <bd0d47c27bd84106afaaecc2c74cdb94>:0)
```

Reply:
Looks like this was the same issue preventing NinjaPedroX from being able to relog on also.  It looks like the SubNameInput isn't always in the right state.  I was able to apply a fix that works better.